### ISSUE TITLE
Two servers run at one time if WDS installed both globally and as project dependency

### ISSUE THREAD

**Issue by pttsky** [webpack/webpack-dev-server#1125]
If I have WDS installed globally and then in some project folder as dependency listed in `package.json`, I got two servers running at one time (See ` How can we reproduce the behavior`).

The issue has gone I run `npm uninstall webpack-dev-server` in the project folder and leave it only installed globally.

I realize that WDS is actually rather a command line tool than a package so it makes no sense listing it as dependency so I just describe the situation and you'll make the decisions.

* Operating System: Ubuntu 16.04.2 LTS
* Node Version: v6.11.3
* NPM Version: 3.10.10
* webpack Version: 3.6.0
* webpack-dev-server Version: 2.9.1

- [x] This is a **bug**
- [ ] This is a **feature** request
- [ ] This is a **modification** request

### Code

```js
  // webpack.config.js

var path = require('path')
  , HtmlWebpackPlugin = require('html-webpack-plugin')
  ;

module.exports = function(env, argv) {
  return {
    context: path.resolve(__dirname, 'src'),
    devServer: {
      contentBase: path.resolve(__dirname, 'build')
    },
    entry: './main.js',
    output: {
      filename: 'main.[hash].js',
      path: path.resolve(__dirname, 'build')
    },
    plugins: [
      new HtmlWebpackPlugin()
    ]
  }
}
```

### Expected Behavior
```
Project is running at http://localhost:8080/
webpack output is served from /
Content not from webpack is served from /path/to/empty-project/build
Hash: 78a1b6d6350cbce5ae42
Version: webpack 3.6.0
Time: 1685ms
```

### Actual Behavior
```
Project is running at http://localhost:8080/
webpack output is served from /
Content not from webpack is served from /path/to/empty-project/build
Project is running at http://localhost:8081/
webpack output is served from /
Content not from webpack is served from /path/to/empty-project/build
Hash: 234350c7d9956dc66340
Version: webpack 3.6.0
Time: 3607ms
```

###  How can we reproduce the behavior?
Run following:
```
sudo npm i -g webpack webpack-dev-server
mkdir empty-project
cd empty-project
npm init 
npm i --save webpack webpack-dev-server html-webpack-plugin
mkdir src
echo "window.onload = function() { alert('hi there'); }" > src/main.js
webpack-dev-server
```

**Comment by shellscape** [comment]
I believe there is an intermittent problem with the global vs local install detection with the code [here](https://github.com/webpack/webpack-dev-server/blob/master/bin/webpack-dev-server.js#L18). however, the workaround (and preferred way) is to install webpack-dev-server locally, and run webpack-dev-server via an npm script defined in your package.json file. npm will always use a local binary over a global binary from a module, in the event one is installed globally and locally.

For example:

```json
"scripts:" {
  "start:dev": "webpack-dev-server --open --config webpack.config.js"
}
```
and
```console
npm run start:dev
```

That block of code in question (which I linked to) has been replaced and a better method introduced in the [`beta` branch](https://github.com/webpack/webpack-dev-server/blob/beta/cli.js#L19), which will be available for testing in a few weeks. In the meantime, you can use the workaround which will definitely avoid the issue you're experiencing.

**Comment by johnnyreilly** [comment]
Hi,

I bumped into the "Two servers run at one time" issue with 2.9.1.  However in my case there was no globally installed `webpack-dev-server`.  Strangely this was an issue on Windows 10 but **not** on Windows 7.

I'm able to workaround by rolling back to 2.7.1.  I wanted to share since I think there are other issues here that haven't been picked up.

I also found that 2.9.1 didn't generally recognise the `devServer` option and I had to move config into the CLI
